packages/common/src/permissions/_internal/checkAssertion.ts

```diff
@@ -0,0 +1,232 @@
+ * @param context
+ * @returns
+ */
+export function checkAssertion(
+  assertion: IPolicyAssertion,
+  entity: Record<string, any>,
+  context: IArcGISContext
+): IPolicyCheck {
+  let response: PolicyResponse = "granted";
+  // construct a hash to look up properties in
+  const lookupHash = { entity, context };
+  // get the property value
+  const propertyLookup = parseProperty(assertion.property);
+  const propValue = getProp(lookupHash, propertyLookup.path);
+
+  // If prop is undefined, then the assertion fails
+  if (propValue === undefined) {
+    response = "assertion-property-not-found";
+  }
+
+  let val = assertion.value;
+  if (typeof val === "string" && val.indexOf(":") > -1) {
+    const valueLookup = parseProperty(assertion.value);
+    val = getProp(lookupHash, valueLookup.path);
+    if (val === undefined) {
+      response = "assertion-property-not-found";
+    }
+  }
+
+  // if we have the two values, we can make the assertion
+  if (propValue !== undefined && val !== undefined) {
+    // TODO: Should these fns return IPolicyCheck? or just the response?
+    // TODO: Should these fns inspect the assertion internally? or switch case here?
+    switch (assertion.assertion) {
+      case "eq":
+      case "neq":
+        response = equalityAssertions(assertion, propValue, val);
+        break;
+      case "contains":
+      case "without":
+        response = arrayAssertions(assertion, propValue, val);
+        break;
+      case "included-in":
+        response = includeAssertions(assertion, propValue, val);
+        break;
+      case "gt":
+      case "lt":
+        response = rangeAssertions(assertion, propValue, val);
+        break;
+      case "is-group-admin":
+      case "is-group-member":
+      case "is-group-owner":
+        response = groupAssertions(assertion, propValue, val, context);
+        break;
+    }
+  }
+
+  const result: IPolicyCheck = {
+    name: `assertion: ${assertion.property} ${assertion.assertion} ${assertion.value}`,
+    value: propValue,
+    code: getPolicyResponseCode(response),
+    response,
+  };
+
+  return result;
+}
+
+function groupAssertions(
+  assertion: IPolicyAssertion,
+  propValue: any,
+  val: any,
+  context: IArcGISContext
+): PolicyResponse {
+  let response: PolicyResponse = "granted";
+  const userGroups = context.currentUser.groups || [];
+
+  // Default the groups to all groups the user is a member of
+  let groups = mapBy("id", userGroups);
+  let failResponse: PolicyResponse = "user-not-group-member";
+
+  // if they are manage, filter to those groups
+  if (assertion.assertion === "is-group-admin") {
+    failResponse = "user-not-group-manager";
+    groups = userGroups.reduce((acc, grp) => {
+      if (grp.userMembership.memberType === "admin") {
+        acc.push(grp.id);
+      }
+      // if they are an owner, they are also a manager
+      if (grp.userMembership.memberType === "owner") {
+        acc.push(grp.id);
+      }
+      return acc;
+    }, []);
+  }
```

Comment:
nit: it would be nice if these reduce functions were packaged up into a well-named function in order to hide implementation details.